systems-management/nisysmgmt.yml

```diff
@@ -0,0 +1,2030 @@
+    title: Create Job Response
+    description: Response describing a job that was sent to systems.
+    type: object
+    properties:
+      jid:
+        description: The ID of the created job.
+        type: string
+        example: 20190211135234448026_28320
+      tgt:
+        description: Array of strings representing system IDs.
+        type: array
+        items:
+          type: string
+          example: VMware_Virtual_Platform--SN-VMware-56_4d_30_26_4f_c9_5c_05-b7_68_d1_cd_99_6e_22_16--MAC-02-01-02-03-04-08
+      fun:
+        description: A string or an array of strings each representing an execution module function.
+        type: object
+        example:
+          - pkg.remove
+          - pkg.install
+      arg: 
+        description: An array of variable type elements that are arguments to the function specified by the "fun" property. If multiple functions are specified, an array of arrays in equivalent order each representing the arguments to a respective function. If multiple functions are specified and at least one requires arguments, an array must be provided for each function, even if it is empty. Keyword arguments may be provided as a parameter of type object with at least one property of type boolean, "__kwarg__", that is evaluated to true.
+        type: array
+        items:
+          type: object
+          example:
+            - __kwarg__: true
+              pkgs:
+              - acl: ''
+        example:
+          - - __kwarg__: true
+              pkgs:
+              - acl: ''
+          - - __kwarg__: true
+              pkgs:
+              - acl-dbg: 1.0.0-0
+      metadata:
+        description: Additional information passed along with the job.
+        type: object
+        additionalProperties:
+          type: object
+        example:
+          queued: true
+          timeout: 600
+          refresh_minion_cache:
+            grains: true
+    required: [jid, tgt, fun]
+
+  RegisterSystemsRequest:
+    description: Request for registering systems.
+    title: Register Systems Request
+    type: object
+    properties:
+      systems_credentials:
+        description: An array of objects representing system credentials.
+        type: array
+        items:
+          $ref: '#/definitions/SystemCredentials'
+      activate:
+        type: boolean
+        description: Whether to activate the systems or not.
+        default: false
+        example: true
+    required: [systems_credentials]
+
+  SystemCredentials:
+    description: Credentials of a system.
+    title: System Credentials
+    type: object
+    properties:
+      hostname:
+        type: string
+        description: Required if ip_address is not provided. A string representing the hostname of the system to register.
+        example: testvm
+      ip_address:
+        type: string
+        description: Required if hostname is not provided. A string representing the IP address of the system to register.
+        example: 1.1.1.1
```

Comment:
Other places use 10.10.10.10 as the example ip_address.  I personally like that better than 1.1.1.1.